### DISCUSSION TITLE
Patch replace is taking half minute

### DISCUSSION THREAD

**Discussion by jj80** [open-policy-agent/community#142] [Community]
We are using opa 0.36 and noticed long response times in patch replace operations.
Here is sample of request
{
    "client_addr": "XXXX",
    "level": "info",
    "msg": "Received request.",
    "req_body": "[{\"op\":\"replace\",\"path\":\"Acls/XYZ\",\"value\":{\"X\":\"A\",\"XYZ\":[\"12345\"],\"LIST\":[\"3456\",\"4567\",\"5678\",\"6789\",\"789\",\"890\",\"9012\",\"01111\",\"55558\",\"3434343\",\"5453422\",\"43242345\",\"5354353\",\"5345345\",\"5646546\",\"5345345\",\"87974466\",\"4456411\",\"445477\",\"6879911\",\"7842313\",\"5411000\",\"364894\",\"321888\",\"7779912\",\"799\",\"3488931\",\"48941699\",\"971fddfg61\"],\"OOO\":null,\"WER\":false}}]",
    "req_id": 30460,
    "req_method": "PATCH",
    "req_params": {},
    "req_path": "/v1/data/AAA/BB/CCC",
    "time": "2022-03-04T08:48:42Z"
}
Response is like
{
    "client_addr": "XXXX",
    "level": "info",
    "msg": "Sent response.",
    "req_id": 30460,
    "req_method": "PATCH",
    "req_path": "/v1/data/AAA/BB/CCC",
    "resp_body": "",
    "resp_bytes": 0,
    "resp_duration": 25305.964576,
    "resp_status": 204,
    "time": "2022-03-04T08:49:08Z"
}

As you can see response time is 25 secs and for multiple similar requests it simple adds up.
We have about 10k entries.
It looks like it is dependent on value and and number of entries. With more values in LIST delay grows.
Is there anything I can do about this?
Does it mean that till patch is processed I will get outdated data?

**Comment by anderseknert** [comment]
Thanks for reporting this @jj80 👍 I've created an issue for tracking in OPA, over [here](https://github.com/open-policy-agent/opa/issues/4409).

**Reply by jj80** [reply]
Thanks for quick reply and creating ticket.
